### ISSUE TITLE
Masking behavior of concatenation should be documented

### ISSUE THREAD

**Issue by gabbard** [keras-team/keras#14034]
**System information**  
- nothing is applicable except:
   - Keras version: 2.3.1

**Describe the current behavior**  

The behavior of the `Concatenate` layer with respect to masks on its inputs [is not documented](https://keras.io/api/layers/merging_layers/concatenate/).  

**Describe the expected behavior**  

It ought to be documented. If I am reading [the code](https://github.com/keras-team/keras/blob/7a39b6c62d43c25472b2c2476bd2a8983ae4f682/keras/layers/merge.py#L406) correctly, the behavior is what you would hope it is: a position is masked out if it is masked out in any of the entries along the final axis of the concatenation result.  However, it would be nice not to need to check the source to confirm this. :-)

More generally, it would be useful if behavior with respect to masking were a standard part of layer documentation (especially when it is not trivial)

**Code to reproduce the issue**  
n/a

**Other info / logs**  
n/a

**Comment by sushreebarsa** [comment]
@gabbard  Moving this issue to closed status as there has been no recent activity, in case you still face the error please create a new issue.Thanks!

**Comment by gabbard** [comment]
@sushreebarsa: This issue notes a gap in the documentation. That gap still seems to be present.

**Comment by LukeWood** [comment]
Hey there @gabbard, would you be interested in contributing this?

**Comment by Krystal225** [comment]
@LukeWood I want to contribute to this.
can you tell me how do I contribute to documentation

**Comment by LukeWood** [comment]
Hey @Krystal225!  The API doc page linked above is generated from the class docstrings.

You can see this specific example at:
https://github.com/keras-team/keras/blob/master/keras/layers/merge.py#L429

Note we use doctests:

A brief overview... statements that output a value are tested against lines without prepended >>>:
```
>>> 'hello'
'hello'
```
passes

```
>>> 'hello'
'another value'
```
Would fail.  You can use this to demonstrate and validate behavior, as you'll see in https://github.com/keras-team/keras/blob/master/keras/layers/merge.py#L457

Please also be sure to review our Contributors guide!
https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md

Let me know if you need any more information!

**Comment by LukeWood** [comment]
@Krystal225 

here's an official documentation guide:
https://www.tensorflow.org/community/contribute/docs_ref

**Comment by Krystal225** [comment]
I have an issue 
I tried doctest on https://github.com/keras-team/keras/blob/master/keras/layers/merge.py#L429 
for demonstration. the test failed and said:
```
Failed example:
    print(x)
Expected:
    [[[ 0  1  2  3  4]
      [ 5  6  7  8  9]]
     [[10 11 12 13 14]
      [15 16 17 18 19]]]
Got:
    [[[ 0  1  2  3  4]
      [ 5  6  7  8  9]]
    <BLANKLINE>
     [[10 11 12 13 14]
      [15 16 17 18 19]]]
Trying:
    y = np.arange(20, 30).reshape(2, 1, 5)
Expecting nothing
ok
Trying:
    print(y)
Expecting:
    [[[20 21 22 23 24]]
     [[25 26 27 28 29]]]
```

need help on how to fix BLANKLINE

**Comment by Krystal225** [comment]
For testing I needed to import numpy as np at the beginning of merge.py

Should I directly add it to code and send a PR?

**Comment by gabbard** [comment]
@LukeWood : I believe this should be assigned to @Krystal225 instead of me.

**Comment by LukeWood** [comment]
> @LukeWood : I believe this should be assigned to @Krystal225 instead of me.

ah thanks -  I updated it.

**Comment by LukeWood** [comment]
> For testing I needed to import numpy as np at the beginning of merge.py
> 
> Should I directly add it to code and send a PR?

Can you elaborate on what you mean here exactly?  If I remember correctly numpy is pre-imported in doctests (I may be wrong)

**Comment by Krystal225** [comment]
on my system I had to add 
```py
import numpy as np
```
in the beginning of merge.py for doctest to work.
other wise all tests that had .np in their code failed.

@LukeWood 


Also is the BLANKLINE because of which the test at  https://github.com/keras-team/keras/blob/master/keras/layers/merge.py#L429 failed okay?
or we need to correct it?

**Comment by SuryanarayanaY** [comment]
Hello, Thank you for reporting an issue.

We're currently in the process of migrating the new Keras 3 code base from keras-team/keras-core to keras-team/keras.
Consequently, This issue may not be relevant to the Keras 3 code base . After the migration is successfully completed, feel free to reopen this Issue at keras-team/keras if you believe it remains relevant to the Keras 3 code base. If instead this Issue is a bug or security issue in legacy tf.keras, you can instead report a new issue at [keras-team/tf-keras](https://github.com/keras-team/tf-keras/issues), which hosts the TensorFlow-only, legacy version of Keras.

To know more about Keras 3, please read https://keras.io/keras_core/announcement/
